biosys/apps/main/tests/api/test_species_observation.py

```diff
@@ -1462,3 +1533,154 @@ def test_species_name_collected_api_update(self):
         record.refresh_from_db()
         self.assertEqual(record.name_id, new_name_id)
         self.assertEqual(record.species_name, expected_species_name)
+
+
+class TestCompositeSpeciesName(helpers.BaseUserTestCase):
+    """
+    Test for species name composed from Genus, Species, infra_rank, infra_name columns
+    """
+
+    species_facade_class = helpers.LightSpeciesFacade
+
+    @staticmethod
+    def schema_with_4_columns_genus():
```

Comment:
Defined by not tested - I'll leave it to you as to whether to add another test for it.